android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,59 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState()
+
+data class MovieCategoryAsString(val value: String) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryAsStringResource(@StringRes val value: Int) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    var isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+@Composable
+fun MovieCategoryLabel(
+    item: MovieCategoryLabelViewState,
+    modifier: Modifier = Modifier,
+) {
+    Column(modifier = modifier
+        .wrapContentWidth()
+        .height(22.dp)
```

Comment:
Hardcoding the height of the text is generally not a good idea since the text can be enlarged from OS settings. This solution would make the text partially visible and potentially unreadable. Use wrap content instead.